### PR TITLE
Configuration update

### PR DESCRIPTION
**Pull request by joevanwanzeeleKF**
This was a major re-write in order to move away from the json file configuration approach toward the standard vault plugin method of setting configuration on a per-plugin instance basis.

Since this is built in go and needs to have the SHA256 values included in the installation instructions, the release process is currently manual.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,40 @@ certificate authority. After issuance, the certificate is then returned to Hashi
 
 ---
 
+## Table of contents
```
Comment (spbsoluble) [inline README.md:16]: Nice did you have something generate this for you?

File: path_issue_sign.go [hunk 2]
```diff
@@ -1,240 +1,228 @@
 package keyfactor
 
-import (
```
Comment (spbsoluble) [inline path_issue_sign.go:3]: I think `path_certs.go` now implements the functionality this provided. Should we just delete this file?

File: path_fetch.go [hunk 3]
```diff
@@ -1,48 +1,26 @@
 package keyfactor
 
-import (
```
Comment (spbsoluble) [inline path_fetch.go:3]: I think this functionality was moved to `path_ca.go` should we just delete this file?

File: path_revoke.go [hunk 4]
```diff
@@ -1,60 +1,60 @@
 package keyfactor
 
-import (
```
Comment (spbsoluble) [inline path_revoke.go:3]: I think this functionality was moved to `path_cert.go` should we just delete this file?

File: path_certs.go [hunk 5]
```diff
@@ -0,0 +1,618 @@
...
+	}
+	if res.StatusCode != 204 {
+		r, _ := ioutil.ReadAll(res.Body)
+		b.Logger().Info("revocation failed: server returned" + fmt.Sprint(res.StatusCode))
+		b.Logger().Info("error response = " + fmt.Sprint(r))
+		return nil, fmt.Errorf("revocation failed: server returned  %s\n ", res.Status)
+	}
+
+	defer res.Body.Close()
+	_, _ = ioutil.ReadAll(res.Body)
```
Comment (spbsoluble) [inline path_certs.go:468]: What is this line doing if it's not assigning and not catching an error?

File: path_certs.go [hunk 6]
```diff
@@ -0,0 +1,618 @@
...
+	httpReq.Header.Add("content-type", "application/json")
+	httpReq.Header.Add("authorization", "Basic "+encCreds)
+
+	res, err := http.DefaultClient.Do(httpReq)
+	if err != nil {
+		b.Logger().Error("Revoke failed: {{err}}", err)
+		return nil, err
+	}
+	if res.StatusCode != 204 {
+		r, _ := ioutil.ReadAll(res.Body)
```
Comment (spbsoluble) [inline path_certs.go:461]: I get a deprecation warning when I use `ioutil`. This line could be converted to `io.ReadAll`

File: crl_util.go [hunk 7]
```diff
@@ -1,175 +1,16 @@
 package keyfactor
 
-import (
```
Comment (spbsoluble) [inline crl_util.go:3]: I think this functionality was moved to `path_certs.go` should we just delete this file?

**Review by spbsoluble — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (joevanwanzeeleKF) [reply README.md:16]: That would have been nice!  no, I wrote it by hand.

File: path_certs.go (hunk 5 above)
Comment (joevanwanzeeleKF) [reply path_certs.go]: good question.. I think it was leftover from a refactor.  I removed it.

**Review by spbsoluble — APPROVED** [review]

**Pull request merged**